src/main/java/com/grapeup/webapp/testWebApp/controllers/SessionsController.java

```diff
@@ -1,34 +1,43 @@
 package com.grapeup.webapp.testWebApp.controllers;
 
+import com.grapeup.webapp.testWebApp.exceptions.SessionException;
 import com.grapeup.webapp.testWebApp.models.Session;
 import com.grapeup.webapp.testWebApp.repositories.SessionRepository;
 import org.springframework.beans.BeanUtils;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
 
-import java.util.List;
+import java.util.Optional;
 
 @RestController
 @RequestMapping("/api/v1/sessions")
 public class SessionsController {
-    @Autowired
-    private SessionRepository sessionRepository;
+    private final SessionRepository sessionRepository;
+
+    public SessionsController(SessionRepository sessionRepository) {
+        this.sessionRepository = sessionRepository;
+    }
 
     @GetMapping
-    public List<Session> list() {
+    public Iterable<Session> list() {
         return sessionRepository.findAll();
     }
 
     @GetMapping("{id}")
     public Session get(@PathVariable final Long id) {
-        return sessionRepository.getById(id);
+        Optional<Session> existingSession = sessionRepository.findById(id);
+        if(existingSession.isPresent()) {
+            return existingSession.get();
```

Comment:
There is nice method to simplify it by `ResponseEntity.of(<optional goes here>)`, controller advice is a great idea, but a bit overkill for `NOT_FOUND`.